tests/unit/core.js

```diff
@@ -742,6 +742,62 @@ exports.testES6ModulesNamedExportsAffectUnused = function (test) {
   test.done();
 };
 
+exports.testConstRedeclaration = function (test) {
+
+  // consts cannot be redeclared, but they can shadow
+  var src = [
+    "const a = 1;",
+    "const a = 2;",
+    "if (a) {",
+    "  const a = 3;",
+    "}",
+    "for(const a in a) {",
+    "  const a = 4;",
+    "}"
+  ];
+
+  TestRun(test)
+      .addError(2, "const 'a' has already been declared.")
+      .test(src, {
+        esnext: true
+      });
+
+  test.done();
+};
+
+exports.testConstModification = function (test) {
+
+  // consts cannot be modified
+  var src = [
+    "const a = 1;",
+    "const b = { a: 2 };",
+    "a = 2;",
+    "b = 2;",
+    "b.a = 3;",
+    "a++;",
+    "b.a++;",
+    "--a;",
+    "--b.a;",
+    "a += 1;",
+    "a.b += 1;",
+    "const c = () => 1;",
+    "c();",
+    "const d = [1, 2, 3];",
+    "d[0] = 2;",
```

Comment:
add some other operators here --- we should really cover them all.

```js
let x = -<const>;
x = +<const>;
x = <const> + 1;
x = <const> * 2;
x = <const> / 2;
x = <const> % 2;
x = <const> & 1;
x = <const> ^ 1;
```

etc etc